# Lab | Inferential statistics - ANOVA

Note: The following lab is divided in 2 sections which represent activities 3 and 4.

## Part 1

In this activity, we will look at another example. Your task is to understand the problem and write down all the steps to set up ANOVA. After the next lesson, we will ask you to solve this problem using Python.

### Context

Suppose you are working as an analyst in a microprocessor chip manufacturing plant. You have been given the task of analyzing a plasma etching process with respect to changing Power (in Watts) of the plasma beam. Data was collected and provided to you to conduct statistical analysis and check if changing the power of the plasma beam has any effect on the etching rate by the machine. You will conduct ANOVA and check if there is any difference in the mean etching rate for different levels of power.

- State the null hypothesis
- State the alternate hypothesis
- What is the significance level
- What are the degrees of freedom of model, error terms, and total DoF

Data was collected randomly and provided to you in the table as shown: [link to the image - Data](https://education-team-2020.s3-eu-west-1.amazonaws.com/data-analytics/7.05/7.05-lab_data.png)

1. Null hypothesis - changing power of plasma beam has no effect on the etching rate by the machine.
2. Alternate hypothesis - changing power of plasma beam has an effect on the etching rate by the machine.
3. significance level - 95% = 0.05
4. Degrees of Freedom - (ng-1) = 2
5. Error Terms - SSE = ((total1-avg1)^2)+((total3-avg3)^2)+((total3-avg3)^2) = 2266.38
6. Total degrees of freedom (DoF) - (N-ng) = 12

## Part 2

- In this section, use the Python to conduct ANOVA.
- What conclusions can you draw from the experiment and why?

In [1]:
import pandas as pd
import researchpy as rp
import scipy.stats as stats
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [2]:
data = pd.read_excel('W7 - D3 - anova_lab_data.xlsx', sheet_name='data_collected')
data.head()

Power  Etching_Rate
0  160 W          5.43
1  180 W          6.24
2  200 W          8.79
3  160 W          5.71
4  180 W          6.71

In [3]:
model = ols('Etching_Rate ~ C(Power)',data=data).fit()
table = sm.stats.anova_lm(model)
print(table)

            df     sum_sq   mean_sq          F    PR(>F)
C(Power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN


In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Power         15 non-null     object 
 1   Etching_Rate  15 non-null     float64
dtypes: float64(1), object(1)
memory usage: 368.0+ bytes


In [5]:
rp.summary_cont(data['Etching_Rate'])

Variable     N    Mean      SD      SE  95% Conf.  Interval
0  Etching_Rate  15.0  6.7827  1.2286  0.3172     6.1023    7.4631

In [7]:
rp.summary_cont(data['Etching_Rate'].groupby(data['Power']))

N   Mean      SD      SE  95% Conf.  Interval
Power                                               
160 W  5  5.792  0.3199  0.1431     5.3948    6.1892
180 W  5  6.238  0.4343  0.1942     5.6987    6.7773
200 W  5  8.318  0.6696  0.2995     7.4866    9.1494

In [8]:
stats.f_oneway(data['Etching_Rate'][data['Power'] == '160 W'],
               data['Etching_Rate'][data['Power'] == '180 W'],
               data['Etching_Rate'][data['Power'] == '200 W'])

F_onewayResult(statistic=36.87895470100505, pvalue=7.506584272358903e-06)

Looking at the F-Table, F-Critical=3.8853. Taking into account that F-Statistic(36.88) > F-Critical(3.8853), we can reject the null hypothesis and accept that changing the power of the plasma beam has an effect on the etching rate by the machine.
Furthermore, the P-Value(7.5*10^-6) << Significance Level(0.05), further strengthening the position to reject the null hypothesis.